### PR TITLE
Load the update information asyncroniously to fix the page load speed

### LINKED ISSUE #15851: Listing of apps takes too much time.
### Steps to reproduce

1. Install OC 8.1alpha2.
2. List apps enabled or disabled.

### Expected behaviour
Apps are listed quickly.

### Actual behaviour
Apps takes a lot of time to be listed. ~1 minute.

### Server configuration
**Operating system**:
Ubuntu 14.04 LTS

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:** (see ownCloud admin page)
{"installed":true,"maintenance":false,"version":"8.1.0.3","versionstring":"8.1 alpha 2","edition":"Enterprise"}

**Updated from an older ownCloud or fresh install:**
Fresh

**List of activated apps:**
Default in enterprise version.

**The content of config/config.php:**

```

```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** 
no

**Logs**
It is plenty of lots of the same message: 
```
{"reqId":"lPE6aoSNmwG1J1hl1Vvn","remoteAddr":"10.40.40.46","app":"core","message":"Could not get application: cURL error 28: Resolving timed out after 5524 milliseconds","level":3,"time":"2015-04-24T08:51:26+00:00"}
```
### Client configuration

**browser**
Chrome 41

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #15851 

@DeepDiver1975 please rate 8.1 vs 8.2

### CODE REVIEW

**Comment by schiesbn** [comment]
much faster now and the code looks good :+1:

File: settings/js/apps.js [hunk 1]
```diff
@@ -123,6 +123,20 @@ OC.Settings.Apps = OC.Settings.Apps || {
...
+				$.ajax(OC.generateUrl('settings/apps/list?category={categoryId}&includeUpdateInfo=1', {
+					categoryId: categoryId
+				}), {
+					type: 'GET',
+					success: function (apps) {
+						_.each(apps.apps, function(app) {
+							if (app.update) {
+								var $update = $('#app-' + app.id + ' .update');
+								$update.removeClass('hidden');
+								$update.val(t('settings', 'Update to {version}', {version: app.update}));
```
Comment (nickvergessen) [inline settings/js/apps.js]: Should reuse `$l->t('Update to %s', array('{{update}}'))` from the php template here:
```js
$update.val(t('settings', 'Update to %s').replace(/%s/g, app.update));
```

**Comment by nickvergessen** [comment]
@SergioBertolinSG @jnfrmarks

File: settings/js/apps.js (hunk 1 above)
Comment (nickvergessen) [inline settings/js/apps.js]: Will push after jenkins

File: settings/js/apps.js [hunk 2]
```diff
@@ -123,6 +123,20 @@ OC.Settings.Apps = OC.Settings.Apps || {
...
 			complete: function() {
 				$('#apps-list').removeClass('icon-loading');
+				$.ajax(OC.generateUrl('settings/apps/list?category={categoryId}&includeUpdateInfo=1', {
+					categoryId: categoryId
+				}), {
+					type: 'GET',
+					success: function (apps) {
+						_.each(apps.apps, function(app) {
+							if (app.update) {
+								var $update = $('#app-' + app.id + ' .update');
```
Comment (LukasReschke) [inline settings/js/apps.js]: Can we escape this just for the sake of we have no idea what the remote server returns?

File: settings/js/apps.js (hunk 1 above)
Comment (LukasReschke) [inline settings/js/apps.js]: PUSH IT ;-)

Comment (nickvergessen) [inline settings/js/apps.js]: done

**Comment by nickvergessen** [comment]
Second review? @MorrisJobke @SergioBertolinSG @jnfrmarks @LukasReschke

File: settings/js/apps.js [hunk 3]
```diff
@@ -123,6 +123,20 @@ OC.Settings.Apps = OC.Settings.Apps || {
...
+								$update.removeClass('hidden');
+								$update.val(t('settings', 'Update to %s').replace(/%s/g, app.update));
```
Comment (LukasReschke) [inline settings/js/apps.js]: Can't we use `t('settings', 'Update to {version}', {version: 'foo'})` here?

Comment (nickvergessen) [inline settings/js/apps.js]: We can, but that will break translations, which I would like to avoid so close to the RC

Comment (nickvergessen) [inline settings/js/apps.js]: Also see https://github.com/owncloud/core/pull/16856#discussion_r32103958 (same discussion in reverse way ;) :P )

Comment (LukasReschke) [inline settings/js/apps.js]: Pfff - people shall learn English. True.

**Comment by LukasReschke** [comment]
Works. Code looks good too :+1:

**Pull request merged**